src/MainUI.cpp

```diff
@@ -54,15 +59,39 @@ void MainUI::createUI(Wt::WContainerWidget *parent) {
        pugi::xml_node menu_items_menu_node = menu_items_node.child("menu");
        while( menu_items_menu_node ) {
           /* submenu creation */
-          Wt::WPopupMenu *nextLevel = new Wt::WPopupMenu();
+          MenuItem *nextLevel = new MenuItem();
           firstLevel->addMenu(menu_items_menu_node.child_value("label"), nextLevel);
           /* iteration over second occurance of items */
           pugi::xml_node menu_items_menu_items_node = menu_items_menu_node.child("items");
           while ( menu_items_menu_items_node ) {
              /* iteration over internal occurance of menuitem */
              pugi::xml_node menu_items_menu_items_menuitem_node = menu_items_menu_items_node.child("menuitem");
              while( menu_items_menu_items_menuitem_node ) {
-                nextLevel->addItem(menu_items_menu_items_menuitem_node.child_value("label"));
```

Comment:
MenuItem *item = new MenuItem (  name of grass command) 
this name of grass command will be passed to m_grassModuleName.
you can get them later in them->triggered method